### ISSUE TITLE
Overfitting in simple LSTM

### ISSUE THREAD

**Issue by bkj** [fchollet/keras#3269]
I'm training the following mode, predicting 1 out of 100 labels from short texts (>= 30 words):

```
model = Sequential()
model.add(Embedding(input_dim=100000, output_dim=128)) # 100000 is the vocab size, 128 is arbitrary
model.add(LSTM(output_dim=128)) # 128 is arbitrary
model.add(Dropout(0.5))
model.add(Dense(100, activation='softmax')) # 100 class prediction task
model.compile(loss='categorical_crossentropy', optimizer='rmsprop', metrics=['accuracy'])

fitist = model.fit(
    sX_train, sy_train,  # 6M records
    verbose = True, 
    nb_epoch = 20,
    batch_size = 256, # 256 is arbitrary
    validation_data = (sX_valid, sy_valid), # 1M records
    shuffle = True
)
```

The resulting loss and accuracy plots look like:

<img width="746" alt="screen shot 2016-07-20 at 10 28 47 am" src="https://cloud.githubusercontent.com/assets/6086781/16990100/ca35957a-4e64-11e6-9791-62fa7660bb1e.png">
<img width="744" alt="screen shot 2016-07-20 at 10 28 52 am" src="https://cloud.githubusercontent.com/assets/6086781/16990101/ca372e62-4e64-11e6-99c2-2a7f4d18f44e.png">

Obviously there's some overfitting going on.  I've been previously told that the solution to this is to increase regularization, usually via dropout.  But I'm wondering where to increase the dropout?  In the Embedding layer?  In the LSTM layer? Alternatively, is there something else (like changing the optimizer or the learning rate) that could help address this?

Thanks
Ben

**Comment by bkj** [comment]
I subsequently tried with the modification:

```
...
model.add(Embedding(input_dim=100000 + 1, output_dim=128, dropout=0.5, mask_zero=True)) 
...
```

which yielded validation/loss curves that look like:

<img width="774" alt="screen shot 2016-07-21 at 10 27 07 am" src="https://cloud.githubusercontent.com/assets/6086781/17026348/cd90fbdc-4f2d-11e6-97e3-48f025eb432d.png">
<img width="754" alt="screen shot 2016-07-21 at 10 27 13 am" src="https://cloud.githubusercontent.com/assets/6086781/17026349/cd922e44-4f2d-11e6-907d-06814805712d.png">

So that seems like too much regularization  -- the training metrics plateau at worse levels than the validation metrics.  I'm guessing this indicates either too much dropout, or dropout applied at the wrong place.

**Comment by bearpaw** [comment]
@bkj Hi, I get the same problem on my dataset. I have not tried dropout yet. Have you solved the problem?

**Comment by taoxuan** [comment]
Has anyone solved the problem？I almost had the same problem. If I don't add dropout to the LSTM cell, it'll overfit greatly. However, if I add dropout to the LSTM cell, the loss on the training set won't drop which means the learning is not so good. I don't know what to do. Can anyone help me out?

**Comment by aparecidovieira** [comment]
How about use some BN and change the optimizer to ADAM.

**Comment by aviatorBeijing** [comment]
0.5 for the dropout might be too high. Try reduce the dropout to 0.2~0.3.
